src/parser/druid/restoration/modules/items/azeritetraits/RampantGrowth.js

```diff
@@ -0,0 +1,141 @@
+ 2. For every Regrowth healing event, if the target has LB credit the entire Regrowth HoT to Rampant Growth.
+ 3. For every healing event, if the target has LB and Regrowth credit one stack of mastery healing to Rampant Growth.
+ 4. For every Regrowth cast, if the target has LB "ban" all credits from point 2 and 3 for the next 12 seconds.
+ */
+
+class RampanthGrowth extends Analyzer{
+  static dependencies = {
+    statWeights: StatWeights,
+    mastery: Mastery,
+    combatants: Combatants,
+  };
+
+  healingFromHotIncrease = 0;
+  healingFromFreeRegrowthHoT = 0;
+  healingFromFreeRegrowthMastery = 0;
+  avgItemLevel = 0;
+  traitLevel = 0;
+  healingIncreasePerTick = 0;
+  banPeriod = 0;
+  discountHealing = 0;
+  directRegrowthCastsOnLB = 0;
+
+  constructor(...args){
+    super(...args);
+    this.active = this.selectedCombatant.hasTrait(SPELLS.RAMPANTH_GROWTH_TRAIT.id);
+    if(this.active) {
+      this.avgItemLevel = this.selectedCombatant.traitsBySpellId[SPELLS.RAMPANTH_GROWTH_TRAIT.id]
+        .reduce((a, b) => a + b) / this.selectedCombatant.traitsBySpellId[SPELLS.RAMPANTH_GROWTH_TRAIT.id].length;
+      this.traitLevel = this.selectedCombatant.traitsBySpellId[SPELLS.RAMPANTH_GROWTH_TRAIT.id].length;
+      this.healingIncreasePerTick = this.selectedCombatant.traitsBySpellId[SPELLS.RAMPANTH_GROWTH_TRAIT.id]
+        .reduce((sum, rank) => sum + calculateAzeriteEffects(SPELLS.RAMPANTH_GROWTH_TRAIT.id, rank)[0], 0);
+    }
+  }
+
+  on_byPlayer_cast(event) {
+    if(event.ability.guid !== SPELLS.REGROWTH.id) {
+      return;
+    }
+    const combatant = this.combatants.players[event.targetID];
+    if (!combatant) {
+      return;
+    }
+
+    // If regrowth was cast on the LB target, disregard regrowth base HoT/mastery healing on that target for 12 seconds.
+    if(combatant.hasBuff(SPELLS.LIFEBLOOM_HOT_HEAL.id, event.timestamp, 200, null, this.owner.playerId)) {
+      this.directRegrowthCastsOnLB++;
+      this.banPeriod = event.timestamp + REGROWTH_DURATION;
+    } else {
+      this.banPeriod = 0;
+    }
+  }
+
+  on_byPlayer_heal(event) {
+    const spellId = event.ability.guid;
+    const combatant = this.combatants.players[event.targetID];
+
+    // If target has LB and Regrowth and is not marked under the ban period, credit one stack of mastery healing given that the heal scales with mastery.
+    if(combatant
+      && combatant.hasBuff(SPELLS.LIFEBLOOM_HOT_HEAL.id, event.timestamp, 200, null, this.owner.playerId)
+      && combatant.hasBuff(SPELLS.REGROWTH.id, event.timestamp, 200, null, this.owner.playerId)
+      && getSpellInfo(spellId).mastery) {
+      if(this.banPeriod < event.timestamp) {
+        const oneMasteryStackHealing = this.mastery.decomposeHeal(event).oneStack;
+        this.healingFromFreeRegrowthMastery += (oneMasteryStackHealing || 0);
+      } else {
+        this.discountHealing +=this.mastery.decomposeHeal(event).oneStack;
```

Comment:
Is there a reason for the second branch not to default to 0?

Reply:
Thanks, did a refactoring.